### PR TITLE
Add coverage enable/disable option for meson

### PR DESCRIPTION
**Pull request by ArtiomDivak**
[WIP]
added coverage option for the meson build.

Related: https://github.com/containers/bluechi/issues/397
Signed-off-by: Artiom Divak <adivak@redhat.com>

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -27,6 +27,13 @@ if api_bus == 'user'
    common_cflags += '-DUSE_USER_API_BUS'
 endif
 
+add_project_link_arguments('-lgcov', language : 'c')
```
Comment (engelmi) [inline meson.build]: You can move this into `if` since we only need this linker argument when building with coverage.

File: meson_options.txt [hunk 2]
```diff
@@ -6,4 +6,8 @@ option('man', type : 'combo',
     choices : ['true', 'false'], value : 'true',
     description : 'Build and install man pages')
 
+option('set_coverage', type : 'combo',
```
Comment (engelmi) [inline meson_options.txt]: `set_coverage` sounds more like a command. How about `build_with_coverage`?

File: bluechi.spec.in [hunk 3]
```diff
@@ -203,7 +203,7 @@ API description and manually written code to simplify recurring tasks.
 %autosetup
 
 %build
-%meson -Dapi_bus=system
+%meson -Dapi_bus=system -Dset_coverage=true
```
Comment (sandrobonazzola) [inline bluechi.spec.in]: this should be under a conditional. You can see how to do that here: https://github.com/containers/bluechi/pull/443/files

Comment (mwperina) [reply bluechi.spec.in]: Please move all spec changes into the following commit, in this commit we only need a way to enable/disable building with coverage support for meson

File: meson.build [hunk 4]
```diff
@@ -27,6 +27,13 @@ if api_bus == 'user'
...
+# Set option to get coverage collection
+set_coverage = get_option('set_coverage')
```
Comment (mwperina) [inline meson.build]: I'd probably rename the meson variable to `coverage_enabled`

File: meson.build (hunk 1 above)
Comment (mwperina) [reply meson.build]: Not *you can*, but *you need to* is the right term :-;

File: meson_options.txt (hunk 2 above)
Comment (mwperina) [reply meson_options.txt]: Maybe `with_coverage` would be enough

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — APPROVED** [review]
LGTM

**Review by engelmi — APPROVED** [review]
LGTM

**Pull request merged**